cromwell_tools/_cromwell_api.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime, timedelta
+import time
+
+import requests
+import requests_mock
+
+from tenacity import retry, wait_exponential, stop_after_delay
+from ._utilities import validate_cromwell_label, prepare_workflow_manifest
+
+
+class WorkflowFailedException(Exception):
+    pass
+
+
+class WorkflowUnknownException(Exception):
+    pass
+
+
+# todo functools partial for get, post (set the authenticate commands)
+class CromwellAPI:
+
+    # todo remove this if it is not critical for mocking
+    requests = requests
+
+    _status_endpoint = '/api/workflows/v1/{uuid}/status'
+    _workflow_endpoint = '/api/workflows/v1'
+    _metadata_endpoint = '/api/workflows/v1/{uuid}/metadata'
+    _health_endpoint = '/api/engine/v1'
+
+    _failed_statuses = ['Failed', 'Aborted', 'Aborting']
+
+    @classmethod
+    def status(cls, uuid, auth, **kwargs):
+        return cls.requests.get(
+            auth.url + cls._status_endpoint.format(uuid=uuid), auth=auth.auth, headers=auth.header)
+
+    @classmethod
+    def health(cls, auth, **kwargs):
+        return cls.requests.get(
+            auth.url + cls._health_endpoint, auth=auth.auth, headers=auth.header)
+
+    @classmethod
+    @retry(reraise=True, wait=wait_exponential(multiplier=1, max=10), stop=stop_after_delay(20))
+    def run(cls, auth, wdl_file, inputs_json, options_json=None, inputs2_json=None,
+            dependencies_json=None, collection_name=None, label=None, validate_labels=True,
+            **kwargs):
```

Comment:
kwargs isn't used below - I'd suggest removing it here, similarly for other functions